### ISSUE TITLE
Allow Custom Plugins to use helm variables via the GUI

### ISSUE THREAD

**Issue by BMonsalvatge** [argoproj/argo-cd#10474]
# Summary

Currently when using a plugin like the argocd vault plugin, I am no longer able to see the values of the helm chart via the gui and have to set it manually.  Ref: https://github.com/argoproj-labs/argocd-vault-plugin/issues/385#issuecomment-1230484255

# Motivation

When using a CMP, it'd be nice if we could still have the default features of the way helm works in argocd.

# Proposal

Not sure how it should logically function in ArgoCD, but when using a CMP it'd be nice to have the features of helm still available in the GUI.  I am still using helm within ArgoCD to specify the values, they just no longer propagate into the GUI.

**Comment by crenshaw-dev** [comment]
May be partially solved by this: https://github.com/argoproj/argo-cd/pull/9216

**Comment by BMonsalvatge** [comment]
Yea, that would partially solve the issue.  I think the only additional functionality that feels missing is a way to prefill those parameters the way `helm` plugin does by default.  Maybe if there was some way a CMP can express where to get those values from (in the case of helm, pointing to `values.yaml`).

**Comment by crenshaw-dev** [comment]
@BMonsalvatge that's part of #9216! When the plugin sends Argo CD a description of what parameters it accepts, it can generate the list however it wants and pre-fill values which are already set. We designed it this way specifically with Helm in mind.

The feature proposal actually includes a small script which converts a values.yaml to a list of plugin parameters: https://argo-cd.readthedocs.io/en/latest/proposals/parameterized-config-management-plugins/#example-3-simple-helm-cmp

**Comment by BMonsalvatge** [comment]
Thanks for clarifying!  Understanding that now looking over it a second time.  I appreciate the response!

**Comment by crenshaw-dev** [comment]
Yeah I'm pretty excited about the feature, even more excited that it sounds like someone might use it. 😆 We'll get it reviewed and merged in time for 2.6 late next month.
